### PR TITLE
Update docs, Provide the env when loading the model (#327)

### LINKED ISSUE #327: The Lunar Lander example throws a 'NoneType' object error on env variable
The Lunar Lander example in the getting started documentation:
 https://stable-baselines3.readthedocs.io/en/master/guide/examples.html

This example creates the Lunar Lander environment:  env = gym.make('LunarLander-v2') 
It then goes on to performs the Training cycles. 
But when it comes to run the evaluated model, the 
obs = env.reset()    line throw up a  'NoneType' object has no attribute 'reset'

Which suggests that the env variable has been overwritten somehow. 

I can run the model, by simply creating the env = gym.make('LunarLander-v2'),  loading the saved model file, and performing and the  obs = env.reset() and the rendering cycles then operate OK. 

But this does not explain, why the original documented example throw up the NoneType error

### PR DESCRIPTION
**Pull request by lorepieri8**
Small fix to the Docs.

## Description
Provide the env when loading the model, to avoid env=None.

## Motivation and Context
closes #327

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

### CODE REVIEW

File: docs/guide/examples.rst [hunk 1]
```diff
@@ -335,9 +335,6 @@ will compute a running average and standard deviation of input features (it can
   # To demonstrate loading
   del model, env
 
-  # Load the agent
-  model = PPO.load(log_dir + "ppo_halfcheetah")
```
Comment (Miffyli) [inline docs/guide/examples.rst:339]: Model loading should stay here (as it was removed just few lines up).

File: docs/misc/changelog.rst [hunk 2]
```diff
@@ -73,6 +73,7 @@ Documentation:
 - Updated custom policy doc (separate policy architecture recommended)
 - Added a note about OpenCV headless version
 - Corrected typo on documentation (@mschweizer)
+- Provide the environment when loading the model (@lorepieri8)
```
Comment (Miffyli) [inline docs/misc/changelog.rst:76]:
```suggestion
- Update example codes with functioning code (@lorepieri8)
```

File: docs/guide/examples.rst (hunk 1 above)
Comment (lorepieri8) [reply docs/guide/examples.rst:339]: That also works, but then it doesn't load with the env,  which was also deleted few lines up.

Comment (Miffyli) [reply docs/guide/examples.rst:339]: Ah, you are right, my bad! No need for changes here :)

File: docs/misc/changelog.rst (hunk 2 above)
Comment (Miffyli) [inline docs/misc/changelog.rst:76]:
```suggestion
- Provide the environment when loading the model in the examples (@lorepieri8)
```

**Review by Miffyli — APPROVED** [review]
I updated the changelog to be bit more clear, and with that LGTM! Thanks for the PR :)

**Pull request merged**